ostd/src/mm/page_table/boot_pt.rs

```diff
@@ -5,19 +5,79 @@
 //! in order to initialize the running phase page tables.
 
 use alloc::vec::Vec;
+use core::{
+    result::Result,
+    sync::atomic::{AtomicU32, Ordering},
+};
 
 use super::{pte_index, PageTableEntryTrait};
 use crate::{
     arch::mm::{PageTableEntry, PagingConsts},
+    cpu::num_cpus,
     mm::{
         nr_subpage_per_huge, paddr_to_vaddr, page::allocator::PAGE_ALLOCATOR, PageProperty,
         PagingConstsTrait, Vaddr, PAGE_SIZE,
     },
+    sync::SpinLock,
 };
 
 type FrameNumber = usize;
 
-/// A simple boot page table for boot stage mapping management.
+/// The accessor to the boot page table singleton [`BootPageTable`].
+///
+/// The user should provide a closure to access the boot page table. The
+/// function will acquire the lock and call the closure with a mutable
+/// reference to the boot page table as the argument.
+///
+/// The boot page table will be dropped when there's no CPU activating it.
+/// This function will return an [`Err`] if the boot page table is dropped.
+pub(crate) fn with<F>(f: F) -> Result<(), ()>
```

Comment:
How about renaming it to `access_with`?

Reply:
`with_borrow` may be better. Referred to Rust thread local's `LocalKey::with_borrow`.